Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/SortAdapter.cpp

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzFramework/DocumentPropertyEditor/SortAdapter.h>
+
+namespace AZ::DocumentPropertyEditor
+{
+    RowSortAdapter::~RowSortAdapter()
+    {
+    }
+
+    void RowSortAdapter::SetSortActive(bool active)
+    {
+        if (m_sortActive != active)
+        {
+            m_sortActive = active;
+            if (active)
+            {
+                // TODO: This method should genenerate patches to/from source once custom patching is implemented
+                HandleReset();
+
+                // we can use the currently cached tree, if it still exists. NB it will be cleared by DOM changes
+                if (!m_rootNode)
+                {
+                    // TODO: GenerateFullTree();
+                }
+                // TODO: generate RemovalsFromSource patch
+            }
+            else
+            {
+                //TODO: generate PatchToSource patch
+            }
+        }
+    }
+
+    Dom::Value RowSortAdapter::GenerateContents()
+    {
+        if (!m_sourceAdapter)
+        {
+            return Dom::Value();
+        }
+
+        auto filteredContents = m_sourceAdapter->GetContents();
+        if (m_sortActive)
+        {
+            filteredContents = GetSortedValue(filteredContents, m_rootNode.get());
+        }
+        return filteredContents;
+    }
+
+    void RowSortAdapter::HandleReset()
+    {
+        m_rootNode.reset();
+        if (m_sortActive)
+        {
+            GenerateFullTree();
+        }
+        NotifyResetDocument(DocumentResetType::HardReset);
+    }
+
+    void RowSortAdapter::HandleDomChange(const Dom::Patch& patch)
```

Comment:
Looks like `patch` is unused here. Not sure if this will compile.

Reply:
yep! That's what I get for removing stale code without recompiling. Fixed now.